agent/azure-monitor-exporter/src/test/java/com/azure/monitor/opentelemetry/exporter/PreAggregatedMetricsTest.java

```diff
@@ -0,0 +1,363 @@
+                    .hasUnit("ms")
+                    .hasHistogramSatisfying(
+                        histogram ->
+                            histogram.hasPointsSatisfying(
+                                point ->
+                                    point
+                                        .hasSum(150 /* millis */)
+                                        .hasAttributesSatisfying(
+                                            equalTo(SemanticAttributes.HTTP_SCHEME, "https"),
+                                            equalTo(SemanticAttributes.HTTP_HOST, "host"),
+                                            equalTo(SemanticAttributes.HTTP_METHOD, "GET"),
+                                            equalTo(SemanticAttributes.HTTP_STATUS_CODE, 200),
+                                            equalTo(SemanticAttributes.HTTP_FLAVOR, "2.0"))
+                                        .hasExemplarsSatisfying(
+                                            exemplar ->
+                                                exemplar
+                                                    .hasTraceId(spanContext1.getTraceId())
+                                                    .hasSpanId(spanContext1.getSpanId())))));
+
+    listener.onEnd(context1, responseAttributes, nanos(250));
+    MetricTelemetryBuilder builder = MetricTelemetryBuilder.create();
+    MetricData metricData = target;
+    MetricDataMapper.updateMetricPointBuilder(
+        builder, metricData, metricData.getData().getPoints().iterator().next(), true, true);
+    TelemetryItem telemetryItem = builder.build();
+    MetricsData metricsData = (MetricsData) telemetryItem.getData().getBaseData();
+
+    assertThat(metricsData.getProperties())
+        .containsExactlyInAnyOrderEntriesOf(generateExpectedRequestCustomDimensions("http"));
+  }
+
+  private static long nanos(int millis) {
+    return TimeUnit.MILLISECONDS.toNanos(millis);
+  }
+
+  private static Map<String, String> generateExpectedDependencyCustomDimensions(String type) {
+    Map<String, String> expectedMap = new HashMap<>();
+    expectedMap.put(MS_METRIC_ID, DEPENDENCIES_DURATION);
+    expectedMap.put(MS_IS_AUTOCOLLECTED, TRUE);
+    // TODO performance market is updated in HttpClientMetrics
+    //    expectedMap.put(PERFORMANCE_BUCKET, "<250ms");
+    expectedMap.put(OPERATION_SYNTHETIC, FALSE);
+    expectedMap.put(DEPENDENCY_SUCCESS, TRUE);
+    if ("http".equals(type)) {
+      expectedMap.put(DEPENDENCY_TYPE, "http");
+      expectedMap.put(DEPENDENCY_RESULT_CODE, "200");
+    } else {
+      expectedMap.put(DEPENDENCY_TYPE, "grpc");
+    }
+    // TODO test cloud_role_name and cloud_role_instance
+    //    expectedMap.put(
+    //        CLOUD_ROLE_NAME,
+    // telemetryItem.getTags().get(ContextTagKeys.AI_CLOUD_ROLE.toString()));
+    //    expectedMap.put(
+    //        CLOUD_ROLE_INSTANCE,
+    //        telemetryItem.getTags().get(ContextTagKeys.AI_CLOUD_ROLE_INSTANCE.toString()));
```

Comment:
just checking that this is still a TODO?

Reply:
if we can unit test our version of httpclientmetrics, then it's a todo.. if we keep using otel's httpclientmetrics, then no. we want to cover this once we have a long-term solution. i would keep it for now.